### ISSUE TITLE
EDSC-883: Facets disappear when new search term is entered

### ISSUE THREAD

**Issue by earthdata-tasktop** [nasa/edsc-search-test-config-3#41]
Enter 'aerosol' as search term on landing page.  
Then choose Keyword facet search, using Agriculture, Soil, Denitrification  
(narrows to 17 records)  
Then close the Keyword details (click on the ^ key)  
Then type 'modis' in search box at top of screen  
All facets disappear from left sidebar (see screen shot)

This is confusing. Two questions:  
1. Should the existing filter apply to the new search?  
2. If it does apply, shouldn't the filter values be shown on the left side bar, even if the filter results in 0 records found?

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~pquinn]I see you have assigned this to EDSC 1.9.  

Doug told me that it was in EDSC 1.7.4, which is in UAT.   So I verified this yesterday in UAT.  

Is there still work to do in Sprint 1.9?

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Approved during sprint planning

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun][~pquinn][~dshum]The facets still disappear in the case where applying the facets to a new search term results in 0 records found.  

Try searching for MODIS, then do a faceted search using Atmosphere > Aerosol > Aerosol Extinction.  

Then change search term in the 'Type any topic or collection name' box to ASTER.  

The facets disappear, and the screen shows 0 records found.  

This is misleading or confusing to the user - they don't see what filter was applied to ASTER data to result in 0 records.

I think I should move this to Verify Failed until this specific scenario of the problem can be fixed.  Do you all agree?

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~pquinn]I see you have assigned this to EDSC 1.9.  

Doug told me that it was in EDSC 1.7.4, which is in UAT.   So I verified this yesterday in UAT.  

Is there still work to do in Sprint 1.9?

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Note: this should be addressed by default by the upcoming CMR facet changes.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun][~pquinn][~dshum]The facets still disappear in the case where applying the facets to a new search term results in 0 records found.  

Try searching for MODIS, then do a faceted search using Atmosphere > Aerosol > Aerosol Extinction.  

Then change search term in the 'Type any topic or collection name' box to ASTER.  

The facets disappear, and the screen shows 0 records found.  

This is misleading or confusing to the user - they don't see what filter was applied to ASTER data to result in 0 records.

I think I should move this to Verify Failed until this specific scenario of the problem can be fixed.  Do you all agree?

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

*Probably* the fix has not been deployed to UAT yet.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun][~dnewman][~pquinn]I will use the steps in my original problem description as part of my verification scenario, but I also want to test the edge case  

where the second search returns 0 matches if we apply the facets.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

##### Verify

The steps in the description as well as the comment left by [~kathleen_carr] on Dec 14 can be used for verification of this ticket.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~pquinn]I see you have assigned this to EDSC 1.9.  

Doug told me that it was in EDSC 1.7.4, which is in UAT.   So I verified this yesterday in UAT.  

Is there still work to do in Sprint 1.9?

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

In UAT, I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Reclamation/Revegetation.   (I did not see Denitrification as an option, as I had in my original test which resulted in the bug report.  Although I think I saw Denitrication sometimes...)  

My resulting set was 2 granules.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 1 granule.

I then started again, and I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Permafrost.  

My resulting set was 1 granule.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 0 granules.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

*Probably* the fix has not been deployed to UAT yet.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun][~dnewman][~pquinn]I will use the steps in my original problem description as part of my verification scenario, but I also want to test the edge case  

where the second search returns 0 matches if we apply the facets.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

##### Verify

The steps in the description as well as the comment left by [~kathleen_carr] on Dec 14 can be used for verification of this ticket.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~pquinn]I see you have assigned this to EDSC 1.9.  

Doug told me that it was in EDSC 1.7.4, which is in UAT.   So I verified this yesterday in UAT.  

Is there still work to do in Sprint 1.9?

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

In UAT, I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Reclamation/Revegetation.   (I did not see Denitrification as an option, as I had in my original test which resulted in the bug report.  Although I think I saw Denitrication sometimes...)  

My resulting set was 2 granules.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 1 granule.

I then started again, and I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Permafrost.  

My resulting set was 1 granule.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 0 granules.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun][~dnewman][~pquinn]I will use the steps in my original problem description as part of my verification scenario, but I also want to test the edge case  

where the second search returns 0 matches if we apply the facets.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

##### Verify

The steps in the description as well as the comment left by [~kathleen_carr] on Dec 14 can be used for verification of this ticket.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~pquinn]I see you have assigned this to EDSC 1.9.  

Doug told me that it was in EDSC 1.7.4, which is in UAT.   So I verified this yesterday in UAT.  

Is there still work to do in Sprint 1.9?

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

In UAT, I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Reclamation/Revegetation.   (I did not see Denitrification as an option, as I had in my original test which resulted in the bug report.  Although I think I saw Denitrication sometimes...)  

My resulting set was 2 granules.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 1 granule.

I then started again, and I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Permafrost.  

My resulting set was 1 granule.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 0 granules.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

##### Verify

The steps in the description as well as the comment left by [~kathleen_carr] on Dec 14 can be used for verification of this ticket.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~pquinn]I see you have assigned this to EDSC 1.9.  

Doug told me that it was in EDSC 1.7.4, which is in UAT.   So I verified this yesterday in UAT.  

Is there still work to do in Sprint 1.9?

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

In UAT, I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Reclamation/Revegetation.   (I did not see Denitrification as an option, as I had in my original test which resulted in the bug report.  Although I think I saw Denitrication sometimes...)  

My resulting set was 2 granules.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 1 granule.

I then started again, and I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Permafrost.  

My resulting set was 1 granule.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 0 granules.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

In UAT, I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Reclamation/Revegetation.   (I did not see Denitrification as an option, as I had in my original test which resulted in the bug report.  Although I think I saw Denitrication sometimes...)  

My resulting set was 2 granules.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 1 granule.

I then started again, and I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Permafrost.  

My resulting set was 1 granule.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 0 granules.

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

In UAT, I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Reclamation/Revegetation.   (I did not see Denitrification as an option, as I had in my original test which resulted in the bug report.  Although I think I saw Denitrication sometimes...)  

My resulting set was 2 granules.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 1 granule.

I then started again, and I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Permafrost.  

My resulting set was 1 granule.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 0 granules.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

In UAT, I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Reclamation/Revegetation.   (I did not see Denitrification as an option, as I had in my original test which resulted in the bug report.  Although I think I saw Denitrication sometimes...)  

My resulting set was 2 granules.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 1 granule.

I then started again, and I chose 'aerosol' as my search term in the landing page.  

I then performed a keyword facet search, using Agriculture, Soils, and Permafrost.  

My resulting set was 1 granule.

I then typed 'modis' in the white search box at the top left of the screen.  

The facets that I had chosen on the aerosol search remained on the side bar, and were applied to the modis search, resulting in 0 granules.
